ensime_shared/ensime.py

```diff
@@ -109,13 +111,13 @@ def create_client(self, config_path):
 
         This will launch the ENSIME server for the project as a side effect.
         """
-        server_v2 = self.using_server_v2()
-        editor = Editor(self._vim)
-        launcher = EnsimeLauncher(self._vim, config_path, server_v2)
-        if server_v2:
-            return EnsimeClientV2(editor, self._vim, launcher)
+        config = ProjectConfig(config_path)
+        launcher = EnsimeLauncher(self._vim, config)
+
+        if self.using_server_v2:
```

Comment:
Sorry for the delay, couldn't spare time during the week—I created #355 and I'm going to try to wrap this PR up with some tests today.

> the problem I see is that handling websocket messages is currently the main responsibility of the client, so we have a chicken and egg problem here due to the current implementation of the client relying on inheritance to pick the protocol version. IMHO the client should be only one, having all the WS machinery and after shacking hands with the server it should instantiate the proper version of protocol handlers.

Yep, I'm not sure about what's going to work best here yet either. It'd be nice if responsibility for dealing with connection/messages didn't spread, but it still seems to me like mixins are the most manageable way to deal with protocol versions (open to ideas though). From my POV, I think it will be easier to decide what feels right after knocking out #325 and other server lifecycle aspects that will achieve some clean-up refactoring around these parts of the code.

In my separate ensime-py client library effort I began modeling a `Connection` as an explicit object—that could end up being something that *knows* the protocol version for the connection it represents, so maybe that could end up being a clean way for connections to be started and then passed into the constructor of client instances for subsequent communication. That is a long way from being ready or fully thought out, though 😣